pkg/pushsync/pushsync_test.go

```diff
@@ -402,7 +562,14 @@ func createPushSyncNode(t *testing.T, addr swarm.Address, recorder *streamtest.R
 	mockStatestore := statestore.NewStateStore()
 	mtag := tags.NewTags(mockStatestore, logger)
 	mockAccounting := accountingmock.NewAccounting()
-	mockPricer := pricermock.NewMockService()
+
+	headlerFunc := func(h p2p.Headers, a swarm.Address) p2p.Headers {
```

Comment:
not sure about this. since you can already inject a mock headler, you don't need to use production code in `headerutils`, therefore you can also unexport all of the helper functions to parse individual headers from `headerutils` package. I would just inject a mock headler here that already has a predefined behavior